add.go

```diff
@@ -168,10 +168,14 @@ func (b *Builder) Add(destination string, extract bool, options AddAndCopyOption
 			return errors.Wrapf(syscall.ENOENT, "no files found matching %q", src)
 		}
 		for _, gsrc := range glob {
-			srcfi, err := os.Stat(gsrc)
+			esrc, err := filepath.EvalSymlinks(gsrc)
 			if err != nil {
 				return errors.Wrapf(err, "error reading %q", gsrc)
```

Comment:
Please tweak this error message so that we can distinguish it from the next one.